test/lib/ansible_test/_util/controller/sanity/validate-modules/validate_modules/schema.py

```diff
@@ -487,10 +487,17 @@ def check_option_choices(v):
         type_checker, type_name = get_type_checker({'type': v.get('elements')})
     else:
         type_checker, type_name = get_type_checker(v)
+
     if type_checker is None:
         return v
 
-    for value in v_choices:
+    if isinstance(v_choices, dict):
+        # choices are still a list (the keys) but dict form serves to document each choice.
+        iterate = v_choices.keys()
+    else:
+        iterate = v_choices
+
+    for value in iterate:
```

Comment:
Is this change necessary? `for value in v_choices` is equivalent to `for value in v_choices.keys()` if `v_choices` is a dict.

Reply:
python versions don't always agree on that, i played it safe since i didn't see a min python version

Reply:
i think i got it!